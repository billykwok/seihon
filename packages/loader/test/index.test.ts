import path from 'path';
import prettier from 'prettier';
import { describe, expect, test } from '@jest/globals';

import compile from './compiler';

describe('loader', () => {
  test('generates collection', async () => {
    const stats = await compile(path.resolve(__dirname, './seihon.config.js'));
    const output = stats.toJson({ source: true }).modules[0].source as string;
    expect(prettier.format(output, { parser: 'babel' })).toMatchSnapshot();
  }, 30000);
});
