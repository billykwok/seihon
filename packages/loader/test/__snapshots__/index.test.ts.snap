// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader generates collection 1`] = `
"export default [
  {
    title: \\"Hello, world!\\",
    date: \\"Apr 22, 2019\\",
    summary: \\"Below is an example of JSX embedded in Markdown.\\",
    tags: [\\"Introduction\\", \\"Testing\\"],
  },
];
"
`;

exports[`loader transforms MDX 1`] = `
"function _extends() {
  _extends =
    Object.assign ||
    function (target) {
      for (var i = 1; i < arguments.length; i++) {
        var source = arguments[i];
        for (var key in source) {
          if (Object.prototype.hasOwnProperty.call(source, key)) {
            target[key] = source[key];
          }
        }
      }
      return target;
    };
  return _extends.apply(this, arguments);
}

import React from \\"react\\";
import { mdx } from \\"@mdx-js/react\\";
/* @jsx mdx */

export const frontmatter = {
  title: \\"Hello, world!\\",
  date: \\"Apr 22, 2019\\",
  summary: \\"Below is an example of JSX embedded in Markdown.\\",
  tags: [\\"Introduction\\", \\"Testing\\"],
};
const layoutProps = {
  frontmatter,
};
const MDXLayout = \\"wrapper\\";
export default function MDXContent({ components, ...props }) {
  return mdx(
    MDXLayout,
    _extends({}, layoutProps, props, {
      components: components,
      mdxType: \\"MDXLayout\\",
    }),
    mdx(
      \\"h2\\",
      null,
      \`Hello, \`,
      mdx(
        \\"em\\",
        {
          parentName: \\"h2\\",
        },
        \`world\`
      ),
      \`!\`
    ),
    mdx(
      \\"p\\",
      null,
      \`Below is an example of JSX embedded in Markdown. \`,
      mdx(\\"br\\", null),
      \` \`,
      mdx(
        \\"strong\\",
        {
          parentName: \\"p\\",
        },
        \`Try and change
the background color!\`
      )
    ),
    mdx(
      \\"div\\",
      {
        style: {
          padding: \\"20px\\",
          backgroundColor: \\"tomato\\",
        },
      },
      mdx(\\"h3\\", null, \\"This is JSX\\")
    )
  );
}
MDXContent.isMDXComponent = true;
"
`;
